tiledb/sm/array_schema/array_schema.cc

```diff
@@ -80,13 +80,20 @@ class ArraySchemaException : public StatusException {
 /* ****************************** */
 
 ArraySchema::ArraySchema(
-    ArrayType array_type, shared_ptr<MemoryTracker> memory_tracker)
+    ArrayType array_type,
+    shared_ptr<MemoryTracker> memory_tracker,
+    std::optional<std::pair<uint64_t, uint64_t>> timestamp_range)
     : memory_tracker_(memory_tracker)
     , uri_(URI())
     , array_uri_(URI())
     , version_(constants::format_version)
-    , timestamp_range_(std::make_pair(
-          utils::time::timestamp_now_ms(), utils::time::timestamp_now_ms()))
+    , timestamp_range_(
+          timestamp_range.has_value() ? std::make_pair(
+                                            timestamp_range.value().first,
+                                            timestamp_range.value().second) :
+                                        std::make_pair(
+                                            utils::time::timestamp_now_ms(),
+                                            utils::time::timestamp_now_ms()))
```

Comment:
* We should be calling `timestamp_now_ms()` only once.
* Can the affirmative part of the conditional be written as `timestamp_range.value()`?